cmake/Common.cmake

```diff
@@ -28,6 +28,11 @@ foreach(p LIB BIN INCLUDE DATA)
   endif()
 endforeach()
 
+# -----------------------------------------------------------------------------
+# To distinguish between debug and release lib
+# -----------------------------------------------------------------------------
+set(CMAKE_DEBUG_POSTFIX "d")
```

Comment:
Would this also apply on linux/mac builds ?

Reply:
On Linux, if we want the indicate the debug or release configuration, the "-DCMAKE_BUILD_TYPE" option has to be added : 
 - for Debug : "cmake -DCMAKE_BUILD_TYPE=Debug"
 - for Release : "cmake -DCMAKE_BUILD_TYPE=Debug" (or "cmake", Release is the default configuration)
With this new line in Common.cmake(set(CMAKE_DEBUG_POSTFIX "d")) : 
 - for Debug configuration, the generated file is libdgtald.so (on Windows : DGtald.lib)
 - for Release configuration, the generated file is libdgtal.so (on Windows : DGtal.lib)

For Mac I cannot perform any test, but I imagine it's the same thing than on Linux.

Reply:
I know that but was just wondering if the "d" suffix would also be used by cmake on linux for the dynamic lib construction.

Reply:
which is not standard for me

Reply:
(and would imply a tweak of the FindDGtal)

Reply:
What is the issue on windows if you remove this suffix ?

Reply:
Maybe I can add the MSVC test ? 
`if (MSVC)`
`  set(CMAKE_DEBUG_POSTFIX "d")`
`endif (MSVC)`

Reply:
maybe.. but what's wrong for windows without ?

Reply:
On Windows, when I "cmake" DGtal for Visual Studio, the two configurations (Release and Debug) are automatically generated. Under Visual Studio, I can choose the configuration to build and install the library. After installing (make install equivalent), the file C:\Program Files\DGtal\lib\DGtal.lib is created (either in Debug or Release configuration, it's the same thing) (1).

Next to that, when I create a new project on Visual Studio, both of the configurations are automatically created for this new project, and both are useful (one for releasing the project, the other one for debugging) (2).

If my new project references DGtal, on the linking step, the Microsoft compiler works with C:\Program Files\DGtal\lib\DGtal.lib.
But if I created DGtal.lib with the Release configuration of DGtal (in step (1)), my new project, in its Debug configuration, won't be able to link with these library. 
On the other hand, if I created DGtal.lib with the Debug configuration of DGtal (in step (1)), my new project, in its Release configuration, won't be able to link with these library.

The debug version of DGtal.lib (size : 20 Mo) contains more information than the release version (size : 3 Mo), and the configuration (Debug or Release) is tagged on the windows static library files.

I hope my explanation is clear ...

Reply:
That's why I suggest adding a DGtald.lib for the Debug configuration

Reply:
This is weird... on linux/mac, you can build an external program in Debug, linking to a Release library and it is just fine. If you debug your code, you will not have access to the debug symbols in the lib (which was compiled in Release) but it is just fine, you can still compile / run / debug you own code.

You are saying that it is not the case in windows ?

Reply:
I'm doing some tests for an alternative solution (without changing DGtal Cmake file :-) ). But the 2 errors on Visual studio are "_ITERATOR_DEBUG_LEVEL" and "RuntimeLibrary" : "https://docs.microsoft.com/en-us/cpp/error-messages/tool-errors/linker-tools-error-lnk2038?view=vs-2017"

Reply:
You can edit the cmake if you want (only for MSVC). I don't undrestand why VC complains but not a big deal if that fixes the issue

Reply:
I succeeded compiling my project in Debug mode, referencing the Release version of DGtal. 
When creating a new project on Visual Studio, the IDE automatically : 
 - for the Debug configuration : adds a preprocessor macro "_DEBUG" (--> _ITERATOR_DEBUG_LEVEL = 2), and sets the property "UseDebugLibraries" to true (--> RuntimeLibrary : Debug)
 - for the Release configuration : adds a preprocessor macro "NDEBUG" (--> _ITERATOR_DEBUG_LEVEL = 0), and sets the property "UseDebugLibraries" to false (--> RuntimeLibrary : Release)

So compiling in Debug mode, linking to the Release version of DGtal, is possible, changing these configurations, but it's not the default configuration for Visual Studio.

Reply:
Ok, I add the "MSVC" test

Reply:
👌